clippy_lints/src/utils/conf.rs

```diff
@@ -106,6 +107,8 @@ macro_rules! define_Conf {
 
 pub use self::helpers::Conf;
 define_Conf! {
+    /// Lint: MANUAL_NON_EXHAUSTIVE, MANUAL_STRIP, OPTION_AS_REF_DEREF, MATCH_LIKE_MATCHES_MACRO. The minimum rust version that the project supports
+    (msrv, "msrv": Option<String>, None),
```

Comment:
@flip1995 what do you think about having this config as an `Option<semver::Version>` instead of `Option<String>`? That way we won't have to check for errors in the `toml` file as `serde` would throw an error if the version string is invalid. 

The error message would be inconsistent though 
`encountered unexpected token: AlphaNumeric("invalid")`

Reply:
That is possible? Hmm... I think we should implement the error checking on our own, because the error message is quite bad with the `semver` deserialization.

Reply:
@flip1995 yeah I didn't think so too but the `define_Conf` macro looks like it'll work with anything that implements `Deserialize`

Reply:
That's good to know. But I prefer the additional code for a better error message.